sys/net/application_layer/nanocoap/sock.c

```diff
@@ -102,41 +147,91 @@ ssize_t nanocoap_sock_request(nanocoap_sock_t *sock, coap_pkt_t *pkt, size_t len
                 continue;
             }
 
-            break;
+            switch (coap_get_type(pkt)) {
+            case COAP_TYPE_RST:
+                /* TODO: handle different? */
+                return -EBADMSG;
+            case COAP_TYPE_CON:
+                _send_ack(sock, pkt);
+                /* fall-through */
+            case COAP_TYPE_NON:
+            case COAP_TYPE_ACK:
+                /* call user callback */
+                res = cb(arg, pkt);
```

Comment:
```suggestion
                if (cb) {
                    res = cb(arg, pkt);
                }
```

Maybe the callback could be optional?